thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js

```diff
@@ -0,0 +1,450 @@
+/**
+ * Component displaying the entity mapping inside a modal
+ * @module components/modals/entity-mapping-modal
+ * @property {Boolean} showEntityMapping  - Flag toggling the modal view
+ * @property {Object} metric              - primary metric Object
+ * @property {Function} onSubmit          - closure action that handles the submit events
+ * @example
+    {{modals/entity-mapping-modal
+      showEntityMappingModal=showEntityMappingModal
+      metric=(get entities metricUrn)
+      onSubmit=(action "onModalSubmit")
+    }}
+ * @exports entity-mapping-modal
+ * @author yyuen
+ */
+
+import Component from '@ember/component';
+import { get, set, getProperties, setProperties, computed, getWithDefault } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { later } from '@ember/runloop';
+import fetch from 'fetch';
+import { task, timeout } from 'ember-concurrency';
+import { checkStatus, postProps } from 'thirdeye-frontend/utils/utils';
+import { deleteProps } from 'thirdeye-frontend/utils/constants';
+import _ from 'lodash';
+
+const MAPPING_TYPES = [
+  'METRIC',
+  'DIMENSION',
+  'SERVICE',
+  'DATASET',
+  'LIXTAG',
+  'CUSTOM'
+];
+
+export default Component.extend({
+  session: service(),
+  /**
+   * Default selected mapping type
```

Comment:
nit: Missing `@type`. There are other properties that are also missing.

Reply:
travis is passing. Going to add those in the next pr. Thanks!